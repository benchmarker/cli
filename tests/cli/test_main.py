import pathlib
from os import PathLike

import pytest
from syrupy.assertion import SnapshotAssertion


@pytest.fixture
def main_file() -> "PathLike[str]":
    path = pathlib.Path(__file__.replace("tests", "benchmarker"), "..", "__main__.py")
    return path.resolve()


def test_main_runs_main(
    main_file: PathLike[str],
    pytester: "pytest.Pytester",
    snapshot: "SnapshotAssertion",
) -> None:
    result = pytester.runpython(main_file)
    assert snapshot == "\n".join(result.errlines)
