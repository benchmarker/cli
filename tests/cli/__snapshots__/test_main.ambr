# name: test_main_runs_main
  '
  usage: __main__.py command
  __main__.py: error: the following arguments are required: command
  '
---
